src/networking/CommunicatorFactory.java

```diff
@@ -49,31 +71,124 @@ public synchronized static ICommunicator getCommunicator(int port){
 		* checking for communicatorinstance if LanCommunicator is already created or not 
 		*/
 		if(communicatorInstance==null){
-
-			/** 
-			* creating Lan communicator object which is going to be used for communication in future. 
-			*/
-			communicatorInstance=new LanCommunicator(port);
-			logger.log(ModuleID.NETWORKING, LogLevel.SUCCESS, "LanCommunicator object created successfully");
+			/** checking to create type of communicator available for communication */
+			if(getTypeOfCommunicator().equals("INTERNET")){
+				/** creating Internet Communicator object, which is going to be used for full session. */
+				communicatorInstance=new InternetCommunicator(port);
+				logger.log(ModuleID.NETWORKING, LogLevel.SUCCESS, "InternetCommunicator object created successfully");
+			}
+			/** creating Lan communicator object which is going to be used for communication in future.	*/
+			else{
+				communicatorInstance=new LanCommunicator(port);
+				logger.log(ModuleID.NETWORKING, LogLevel.SUCCESS, "LanCommunicator object created successfully");
+			} 
 		}
+
 		else
-			logger.log(ModuleID.NETWORKING, LogLevel.WARNING, "LanCommunicator object already created");
-		/** 
-		* returning new or already created LanCommunicator object 
-		*/
+			logger.log(ModuleID.NETWORKING, LogLevel.WARNING, "ICommunicator object already created");
+		
+		/** returning new or already created ICommunicator object */
 		return communicatorInstance;
 	}
 
-	
+
+	/**
+	* This access the xml file which consists of Server IP address and listening port at server, These details are stored 
+	* by the respective user or client.
+	*
+	* @return String[] which consists of aws server IP address and port
+	*/
+	private static String[] getServerInfo(){
```

Comment:
change from private to default

Reply:
This was not there in the discussed design. I have written this function for my convenience. So Is this new extra requirement to be done??

Reply:
Yeah